snapshot/store/groupcache_store.go

```diff
@@ -120,23 +123,24 @@ type groupcacheStore struct {
 	stat       stats.StatsReceiver
 }
 
-func (s *groupcacheStore) OpenForRead(name string) (io.ReadCloser, error) {
+func (s *groupcacheStore) OpenForRead(name string) (io.ReadCloser, *TTLValue, error) {
 	log.Info("Read() checking for cached bundle: ", name)
 	defer s.stat.Latency(stats.GroupcacheReadLatency_ms).Time().Stop()
 	s.stat.Counter(stats.GroupcacheReadCounter).Inc(1)
 	var data []byte
-	if err := s.cache.Get(nil, name, groupcache.AllocatingByteSliceSink(&data)); err != nil {
-		return nil, err
+	if _, err := s.cache.Get(nil, name, groupcache.AllocatingByteSliceSink(&data)); err != nil {
```

Comment:
Let's actually get the ttl from Get and populate a *TTLValue in the return if it wasn't null. We'll have to use the rest of the TTLValue fields from the TTLConfig

Reply:
Done.